pkg/store/did/store.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package did
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+const (
+	// NameSpace for did store
+	NameSpace = "did"
+
+	didNameKey        = "didname_"
+	didNameKeyPattern = didNameKey + "%s"
+
+	// limitPattern for the iterator
+	limitPattern = "%s~"
+)
+
+// ErrDIDNotFound signals that the entry for the given DID and key is not present in the store.
+var ErrDIDNotFound = errors.New("did not found under given key")
+
+// Store stores did doc
+type Store struct {
+	store storage.Store
+}
+
+type provider interface {
+	StorageProvider() storage.Provider
+}
+
+// New returns a new did store
+func New(ctx provider) (*Store, error) {
+	store, err := ctx.StorageProvider().OpenStore(NameSpace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open did store: %w", err)
+	}
+
+	return &Store{store: store}, nil
+}
+
+// SaveDID saves a did doc.
+func (s *Store) SaveDID(name string, didDoc *did.Doc) error {
```

Comment:
Do we need to store DID Document here? I understand the requirement to save Name and DID mapping. We can always fetch DID Document from VDRI.

Reply:
@sandrask

Reply:
It is true that we can always resolve it. However it is more efficient to store it since the user is supposed to be managing their own did.

Reply:
@troy What is your take on this? Should we store did documents or just dids?

Reply:
I would say the user/holder should have a copy of their own DID documents somewhere locally, if that is the question.

Is this store only for the user’s own DIDs?

Beyond that, It is an interesting question on what functionality belongs to vdri vs here.